butane_codegen/src/lib.rs

```diff
@@ -42,9 +38,8 @@ mod filter;
 /// #[model]
 /// #[table = "posts"]
 /// pub struct Post {
-///   #[auto]
```

Comment:
we could re-implement `#[auto]` so that it injects the new `AutoPk`.
That would remove the largest part that is 'breaking'.

We could also update `#[model]` so that it removes any `state` field automatically, but I guess that isnt worth doing as the code would still fail to compile due to the `use butane::ObjectState` which would be in the code.  Maybe have a dummy `ObjectState` type.  There are deprecation features of Rust we could use to advise people to remove that `use`.

Reply:
fyi, I am happy to attempt this after your PR, if you agree with it.

Reply:
Good suggestion. Let me try to see if I can find time this weekend to do that.